tests/functional/aws-node-sdk/test/object/putVersion.js

```diff
@@ -89,8 +89,13 @@ describe('PUT object with x-scal-s3-version-id header', () => {
                 next => s3.putObject(params, next),
                 next => fakeMetadataArchive(bucketName, objectName, undefined, archive, next),
                 next => getMetadata(bucketName, objectName, undefined, (err, objMD) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    // eslint-disable-next-line no-param-reassign
+                    objMD.legalHold = true;
```

Comment:
why add 'legalHold' to each object?
* we probably want to the test both cases (overwrite with and without legal hold)
* setting legal hold should be done via regular s3 api, not by tweaking the API